scene/animation/animation_tree.h

```diff
@@ -96,10 +97,14 @@ class AnimationNode : public Resource {
 	HashMap<NodePath, bool> filter;
 	bool filter_enabled = false;
 
+	StringName node_name;
```

Comment:
I like that we are unifying name, but why are we adding a new StringName when resources already have a name, accessible as `get_name()` or the  `"resource_name"` property?

Reply:
I didn't use the `resource_name` because I didn't know if it could generate some secondary behavior elsewhere in the editor. The current code didn't do that so I just kept it as it was today.